src/core/thread/network_data_local.cpp

```diff
@@ -58,9 +58,13 @@ Local::Local(ThreadNetif &aThreadNetif):
 ThreadError Local::AddOnMeshPrefix(const uint8_t *aPrefix, uint8_t aPrefixLength, int8_t aPrf,
                                    uint8_t aFlags, bool aStable)
 {
+    ThreadError error = kThreadError_None;
     PrefixTlv *prefixTlv;
     BorderRouterTlv *brTlv;
 
+    VerifyOrExit(memcmp(aPrefix, mNetif.GetMle().GetMeshLocalPrefix(), aPrefixLength / 8),
```

Comment:
It may be worthwhile to have a version that looks like `PrefixMatch(const uint8_ t *aPrefixA, const uint8_t *aPrefixB, uint8_t aMaxLength)` and have `Ip6::PrefixMatch()` use that as well.  Thoughts?